src/lib/svg.js

```diff
@@ -0,0 +1,202 @@
+}
+
+const wrapInSvgTag = (svgObjects, width=400, height=400) => {
+  return `<svg xmlns="http://www.w3.org/2000/svg" width="${width}" height="${height}" viewBox="0 0 ${width} ${height}">` + svgObjects + `</svg>`
+}
+
+const createUseElement = (shape, fill) => {
+  const useElement = document.createElement('use')
+  useElement.setAttribute('href', `#${shape}`)
+  if (fill) {
+    useElement.setAttribute('fill', fill)
+  }
+  return useElement
+}
+
+const createClippedShape = (shape, svgObjects, theme) => {
+  if (shape === 'bg') {
+    return wrapInSvgTag(svgObjects)
+  }
+  const mask = document.createElement('mask')
+  mask.setAttribute('id', 'mask')
+
+  const defs = document.createElement('defs')
+  const symbolSvg = changeToSymbol(getShapeSvg(shape, null, null), 'symbol')
+  defs.appendChild(symbolSvg)
+
+  const maskUse = createUseElement(shape, '#FFF')
+  const outlineUse = createUseElement(shape, null)
+  mask.appendChild(maskUse)
+  defs.appendChild(mask)
+
+  const group = document.createElementNS('http://www.w3.org/2000/svg', 'g')
+  group.setAttribute('mask', 'url(#mask)')
+  group.insertAdjacentHTML('beforeend', svgObjects)
+
+  const parentSvg = document.createElementNS('http://www.w3.org/2000/svg', 'svg')
+  parentSvg.appendChild(defs)
+  parentSvg.appendChild(outlineUse)
+  parentSvg.appendChild(group)
+  parentSvg.setAttribute('width', 400)
+  parentSvg.setAttribute('height', 400)
+  parentSvg.setAttribute('stroke', getStroke(theme))
+  parentSvg.setAttribute('stroke-width', '1%')
+  parentSvg.setAttribute('stroke-linecap', 'round')
+  parentSvg.setAttribute('xmlns', 'http://www.w3.org/2000/svg')
+  return parentSvg.outerHTML
+}
+
+const createGenerativeShape = (shape, artId, theme) => {
+  const artSvgObjects = createArtSvg(artId.replace('generative_', ''), theme, 400, 400)
+  return createClippedShape(shape, artSvgObjects, theme)
+}
+
+const createVoronoiShape = (shape, voronoiId, theme) => {
+  const [seed, splits] = voronoiId.replace('voronoi_', '').split('_')
+  const voronoiSvgObjects = createVoronoiSvg(seed, splits, theme, shape === 'bg' ? 0 : 100)
+  return createClippedShape(shape, voronoiSvgObjects, theme)
+}
+
+const getSvgString = (id) => {
+  const parts = id.split('-')
+  let first = parts[0]
+  let rest = parts.slice(1, parts.length)
+  if (first.includes('_')) {
+    const subParts = first.split('_')
+    first = subParts[0]
+    rest = subParts.slice(1, subParts.length).concat(rest)
+  }
+  switch (first) {
+    case 'basicShape':
+      return createBasicShape(...rest)
+    case 'gradientShape':
+      return createGradientShape(...rest)
+    case 'generativeShape':
+      return createGenerativeShape(...rest)
+    case 'voronoiShape':
+      return createVoronoiShape(...rest)
+    case 'voronoi':
+      return wrapInSvgTag(createVoronoiSvg(...rest))
+    case 'generative':
+      return wrapInSvgTag(createArtSvg(...rest))
+  }
+  return undefined
+}
+
+const cache = new SvgLruCache(256)
```

Comment:
Would be cool to fill this up with the upcoming svg before a trial plays. Either do that or cache every trial's svg at game start.